Author: davilla <davilla@4pi.com>
Date:   Sun Dec 21 17:37:22 2014 -0500
 
    [aml] fixed, do not reset pts_pcrscr if the error is too large, it is just a dvdplayer transient discontinuity
 
diff --git a/xbmc/cores/dvdplayer/DVDCodecs/Video/AMLCodec.cpp b/xbmc/cores/dvdplayer/DVDCodecs/Video/AMLCodec.cpp
index 3435151..9d64aca 100644
--- a/xbmc/cores/dvdplayer/DVDCodecs/Video/AMLCodec.cpp
+++ b/xbmc/cores/dvdplayer/DVDCodecs/Video/AMLCodec.cpp
@@ -1974,7 +1974,11 @@ void CAMLCodec::Process()
           if (abs_error > 0.150)
           {
             // big error so try to reset pts_pcrscr
-            SetVideoPtsSeconds(app_pts);
+            // do not reset if the error is very big,
+            // might just be dvdplayer wonking out on a discontinuity
+            // and we just need to let it settle down.
+            if (abs_error < 10.0)
+              SetVideoPtsSeconds(app_pts);
           }
           else
           {
